### PR TITLE
StudentCourseJoinAuthenticatedAction: NullPointerException in live server #7097

### LINKED ISSUE #7097: StudentCourseJoinAuthenticatedAction: NullPointerException in live server
Received a few of these today
v5.101

```
Error Message: teammates.common.util.Logger severe: Unexpected exception caught by ControllerServlet :
java.lang.NullPointerException
at teammates.ui.controller.StudentCourseJoinAuthenticatedAction.execute(StudentCourseJoinAuthenticatedAction.java:56)
at teammates.ui.controller.Action.executeAndPostProcess(Action.java:361)
at teammates.ui.controller.ControllerServlet.doPost(ControllerServlet.java:64)
at teammates.ui.controller.ControllerServlet.doGet(ControllerServlet.java:41)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:617)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:717)
at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:511)
at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1166)
at com.google.apphosting.utils.servlet.ParseBlobUploadFilter.doFilter(ParseBlobUploadFilter.java:125)
at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
at com.google.apphosting.runtime.jetty.SaveSessionFilter.doFilter(SaveSessionFilter.java:37)
at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
at com.google.apphosting.utils.servlet.JdbcMySqlConnectionCleanupFilter.doFilter(JdbcMySqlConnectionCleanupFilter.java:60)
at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
at com.google.apphosting.utils.servlet.TransactionCleanupFilter.doFilter(TransactionCleanupFilter.java:48)
at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:388)
at org.mortbay.jetty.security.SecurityHandler.handle(SecurityHandler.java:216)
at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:182)
at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:765)
at org.mortbay.jetty.webapp.WebAppContext.handle(WebAppContext.java:418)
at com.google.apphosting.runtime.jetty.AppVersionHandlerMap.handle(AppVersionHandlerMap.java:257)
at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
at org.mortbay.jetty.Server.handle(Server.java:326)
at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:542)
at org.mortbay.jetty.HttpConnection$RequestHandler.headerComplete(HttpConnection.java:923)
at com.google.apphosting.runtime.jetty.RpcRequestParser.parseAvailable(RpcRequestParser.java:76)
at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:404)
at com.google.apphosting.runtime.jetty.JettyServletEngineAdapter.serviceRequest(JettyServletEngineAdapter.java:145)
at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.dispatchServletRequest(JavaRuntime.java:635)
at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.dispatchRequest(JavaRuntime.java:595)
at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.run(JavaRuntime.java:565)
at com.google.tracing.TraceContext$TraceContextRunnable.runInContext(TraceContext.java:453)
at com.google.tracing.TraceContext$TraceContextRunnable$1.run(TraceContext.java:460)
at com.google.tracing.CurrentContext.runInContext(CurrentContext.java:293)
at com.google.tracing.TraceContext$AbstractTraceContextCallback.runInInheritedContextNoUnref(TraceContext.java:319)
at com.google.tracing.TraceContext$AbstractTraceContextCallback.runInInheritedContext(TraceContext.java:311)
at com.google.tracing.TraceContext$TraceContextRunnable.run(TraceContext.java:457)
at com.google.apphosting.runtime.ThreadGroupPool$PoolEntry.run(ThreadGroupPool.java:238)
at java.lang.Thread.run(Thread.java:745)
```

### PR DESCRIPTION
**Pull request by whipermr5**
Fixes #7097

### CODE REVIEW

**Comment by whipermr5** [comment]
Interestingly, `errorCode` is only used exactly twice throughout the entire project - once in `BackDoorLogicTest` and once here.

File: src/main/java/teammates/ui/controller/StudentCourseJoinAuthenticatedAction.java [hunk 1]
```diff
@@ -53,7 +53,7 @@ protected ActionResult execute() throws EntityDoesNotExistException {
             logic.joinCourseForStudent(regkey, account.googleId);
         } catch (JoinCourseException | InvalidParametersException e) {
             // Does not sanitize for html to allow insertion of mailto link
-            if (e.errorCode.equals(Const.StatusCodes.INVALID_KEY)) {
+            if (e.errorCode != null && e.errorCode.equals(Const.StatusCodes.INVALID_KEY)) {
```
Comment (damithc) [inline src/main/java/teammates/ui/controller/StudentCourseJoinAuthenticatedAction.java]: I wonder if [Object.equals()](http://docs.oracle.com/javase/7/docs/api/java/util/Objects.html#equals-java.lang.Object-java.lang.Object-) can be used here 🤔

Comment (whipermr5) [inline src/main/java/teammates/ui/controller/StudentCourseJoinAuthenticatedAction.java]: Ohh TIL of its existence 😅 Alternatively I could do `Const.StatusCodes.INVALID_KEY.equals(e.errorCode)`?

Comment (damithc) [inline src/main/java/teammates/ui/controller/StudentCourseJoinAuthenticatedAction.java]: `Const.StatusCodes.INVALID_KEY.equals(e.errorCode)` sounds good. 👍

**Pull request merged**